### PR TITLE
Add "No explicit `.rb` to `require`" section

### PR DESCRIPTION
**Pull request by fatkodima**

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -5127,6 +5127,29 @@ a = [1, 2, 3].reduce({}) { |h, i| h[i] = i * 17; h }        # good
...
 
+=== No explicit `.rb` to `require` [[no-explicit-rb-to-require]]
+
+Omit the `.rb` extension for filename passed to `require` and `require_relative`.
+
+NOTE: If the extension is omitted, Ruby tries adding '.rb', '.so', and so on to the name
+until found. If the file named cannot be found, a `LoadError` will be raised.
+There is an edge case where `foo.so` file is loaded instead of a `LoadError`
+if `foo.so` file exists when `require 'foo.rb'` will be changed to `require 'foo'`,
+but that seems harmless.
```
Comment (marcandre) [inline README.adoc:5138]: 👍, not sure these 3 lines are necessary

**Comment by marcandre** [comment]
Tangential, but I'm surprised not to see "use `require_relative` for all you own files, and `require` for external dependencies: It's both more expressive and more efficient. @fatkodima would you like to create a PR for this too, or should I?

**Comment by fatkodima** [comment]
@marcandre please, create 😃

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**